### PR TITLE
Add validation for generated json metafiles

### LINKED ISSUE #152: Check for valid JSON on PRs
### Description of the feature you are looking for.

On auto-generated PRs we still should check  for valid JSON in case manual edits were necessary, see #147

### PR DESCRIPTION
**Pull request by kislerdm**
Resolves #152 

## What changed

- Added the program which validates json metadata files for modules and providers by performing a set of checks. It reports all found errors to stderr and terminates with the exit code 1 if the file is invalid, otherwise, it terminates with the exit code 0 without any messages.

## Why do we need it

- To validate correctness of updated json metafiles

#### Provider

image

#### Module

image

### CODE REVIEW

**Comment by kislerdm** [comment]
Hey @janosdebugs @Yantrio @cam72cam! I'd appreciate your quick look over the change to see I'm going in the right direction with the implementation. Thanks!

**Comment by Yantrio** [comment]
I think the implemtnation here is pretty solid, but I think it may be best to have an external type of validation here instead of using the same code that we're generating with.

Maybe it could be as simple as piping it into JQ and ensuring it's valid json?

@janosdebugs you raised the initial issue, when you mention "valid" json, do you mean that it's valid json, or valid to our format and what we expected?

**Comment by ghost** [comment]
@Yantrio as I wrote in the related issue, I would like the JSON to be valid from the perspective of the registry, ideally. However, this may also be covered by #164.

**Comment by kislerdm** [comment]
@janosdebugs @Yantrio Hey folks! Thanks for the feedback. On its basis, I'll remove introduced code and setup a github action which will read modified metafile using `jq`, and implement the content validation as part of #164.

**Comment by ghost** [comment]
To clarify: I'm not unhappy with a Go solution, it feels more robust than shell scripting around it. That being said, I don't know the registry very well, so I'll have to rely on the more experienced colleagues for judgement here.

**Comment by kislerdm** [comment]
@janosdebugs @Yantrio hey guys! The validator is ready for review.

File: .github/workflows/validate-json-module.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          go-version-file: './src/go.mod'
+
+      - name: Build validator
+        working-directory: ./src
+        run: |
+          mkdir -p /tmp/validate
+          go build -o /tmp/validate/run ./cmd/validate/main.go
+          chmod +x /tmp/validate/run
+
+      - name: List updated filed
```
Comment (Yantrio) [inline .github/workflows/validate-json-module.yml]:
```suggestion
      - name: List updated files
```

File: .github/workflows/validate-json-provider.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          go-version-file: './src/go.mod'
+
+      - name: Build validator
+        working-directory: ./src
+        run: |
+          mkdir -p /tmp/validate
+          go build -o /tmp/validate/run ./cmd/validate/main.go
+          chmod +x /tmp/validate/run
+
+      - name: List updated filed
```
Comment (Yantrio) [inline .github/workflows/validate-json-provider.yml]:
```suggestion
      - name: List updated files
```

**Review by Yantrio — APPROVED** [review]
thanks for this @kislerdm , I've added 2 tiny tiny nitpicks but overall im happy to approve.

**Comment by kislerdm** [comment]
@Yantrio good catch, thank you James! 😄

**Comment by kislerdm** [comment]
@janosdebugs Hey Janos! Could you please review the changes? Thanks!

**Review by ghost — APPROVED** [review]
Very nice and thorough work, thank you!

**Pull request merged**